backend/sources/whatsapp/connector/src/test/java/co/airy/core/sources/whatsapp/SendMessageTest.java

```diff
@@ -0,0 +1,190 @@
+                applicationCommunicationChannels,
+                applicationCommunicationMessages,
+                applicationCommunicationMetadata
+        );
+
+        kafkaTestHelper.beforeAll();
+    }
+
+    @AfterAll
+    static void afterAll() throws Exception {
+        kafkaTestHelper.afterAll();
+    }
+
+    @BeforeEach
+    void beforeEach() throws InterruptedException {
+        MockitoAnnotations.openMocks(this);
+        retryOnException(() -> assertEquals(stores.getStreamState(), RUNNING), "Failed to reach RUNNING state.");
+    }
+
+    @Test
+    void canSendMessage() throws Exception {
+        final String conversationId = "conversationId";
+        final String messageId = "message-id";
+        final String failingMessageId = "message-id-failing";
+        final String sourceConversationId = "source-conversation-id";
+        final String channelId = "channel-id";
+        final String token = "token";
+        final String text = "Hello World";
+        final String errorMessage = "message delivery failed";
+        final String whatsappMessageId = "whatsapp message id";
+
+        ArgumentCaptor<SendMessageRequest> payloadCaptor = ArgumentCaptor.forClass(SendMessageRequest.class);
+        ArgumentCaptor<String> tokenCaptor = ArgumentCaptor.forClass(String.class);
+        when(api.sendMessage(payloadCaptor.capture()))
+                .thenReturn(SendMessageResponse.builder()
+                        .messages(List.of(new SendMessageResponse.Message(whatsappMessageId)))
+                        .build())
+                .thenThrow(new ApiException(errorMessage));
+
+        kafkaTestHelper.produceRecords(List.of(
+                new ProducerRecord<>(applicationCommunicationChannels.name(), channelId, Channel.newBuilder()
+                        .setToken(token)
+                        .setSourceChannelId("ps-id")
+                        .setSource("whatsapp")
+                        .setId(channelId)
+                        .setConnectionState(ChannelConnectionState.CONNECTED)
+                        .build()
+                ),
+                new ProducerRecord<>(applicationCommunicationMessages.name(), "other-message-id",
+                        Message.newBuilder()
+                                .setId("other-message-id")
+                                .setSource("whatsapp")
+                                .setSentAt(Instant.now().toEpochMilli())
+                                .setSenderId(sourceConversationId)
+                                .setDeliveryState(DeliveryState.DELIVERED)
+                                .setConversationId(conversationId)
+                                .setChannelId(channelId)
+                                .setContent("{\"text\":\"" + text + "\"}")
+                                .setIsFromContact(true)
+                                .build())
+        ));
+
+        TimeUnit.SECONDS.sleep(5);
```

Comment:
with the following attributes
```
kafka.commit-interval-ms=0
kafka.cache.max.bytes=0
```

You should not need to wait that long, 100ms will be enough? Also there is other way to get around the sleep time?

Reply:
Just tested this and it doesn't make a difference. I think that this time is genuinely spent on catching up with the streams app.